drivers/display/ili9340/ili9340.c

```diff
@@ -0,0 +1,181 @@
+/* Copyright (c) 2017 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ili9340_lcd.h"
+#include <drivers/display/ili9340.h>
+
+#define SYS_LOG_DOMAIN "ILI9340"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ILI9340_LEVEL
+#include <logging/sys_log.h>
+
+#include <gpio.h>
+#include <misc/byteorder.h>
+#include <spi.h>
+
+struct ili9340_data {
+	struct device *spi;
+	struct device *reset_gpio;
+	struct device *command_data_gpio;
+};
+
+#define ILI9340_CMD_ENTER_SLEEP 0x10
+#define ILI9340_CMD_EXIT_SLEEP 0x11
+#define ILI9340_CMD_DISPLAY_OFF 0x28
+#define ILI9340_CMD_DISPLAY_ON 0x29
+#define ILI9340_CMD_COLUMN_ADDR 0x2a
+#define ILI9340_CMD_PAGE_ADDR 0x2b
+#define ILI9340_CMD_MEM_WRITE 0x2c
+
+#define ILI9340_CMD_DATA_PIN_COMMAND 0
+#define ILI9340_CMD_DATA_PIN_DATA 1
+
```

Comment:
move all these macros (and why not the struct ili9340_data too actually) into display_ili9340.h header file